buildkite/resource_pipeline_schedule.go

```diff
@@ -7,250 +7,319 @@ import (
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"uuid": {
+			"uuid": resource_schema.StringAttribute{
 				Computed: true,
-				Type:     schema.TypeString,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"label": {
-				Required: true,
-				Type:     schema.TypeString,
+			"pipeline_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The ID of the pipeline that this schedule belongs to.",
 			},
-			"cronline": {
-				Required: true,
-				Type:     schema.TypeString,
+			"label": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "A label to describe the schedule.",
 			},
-			"commit": {
-				Optional: true,
-				Default:  "HEAD",
-				Type:     schema.TypeString,
+			"cronline": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The cronline that describes when the schedule should run.",
 			},
-			"branch": {
-				Required: true,
-				Type:     schema.TypeString,
+			"branch": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The branch that the schedule should run on.",
 			},
-			"message": {
-				Optional: true,
-				Computed: true,
-				Type:     schema.TypeString,
+			"commit": resource_schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "The commit that the schedule should run on.",
+				Default:             stringdefault.StaticString("HEAD"),
 			},
-			"env": {
-				Optional: true,
-				Type:     schema.TypeMap,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
-				},
+			"message": resource_schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The message that the schedule should run on.",
+			},
+			"env": resource_schema.MapAttribute{
+				ElementType:         types.StringType,
+				Optional:            true,
+				MarkdownDescription: "The environment variables that the schedule should run on.",
 			},
-			"enabled": {
-				Optional: true,
-				Default:  true,
-				Type:     schema.TypeBool,
+			"enabled": resource_schema.BoolAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "Whether the schedule is enabled or not.",
+				Default:             booldefault.StaticBool(false),
 			},
 		},
 	}
 }
 
-func setPipelineScheduleIDFromSlug(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	client := m.(*Client)
+func (ps *pipelineSchedule) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state pipelineScheduleResourceModel
```

Comment:
Creates probably worth having just a single model, since theres no state to begin with, but not a urgent case 👍

Reply:
Will raise a different PR for that improvement.